payment/src/main/java/org/killbill/billing/payment/core/PaymentProcessor.java

```diff
@@ -400,6 +400,12 @@ public boolean apply(final PaymentTransactionModelDao input) {
                         throw new PaymentApiException(ErrorCode.PAYMENT_INVALID_OPERATION, paymentStateContext.getTransactionType(), transactionToComplete.getTransactionStatus());
                     }
 
+                    // If Janitor confirmed the status, just return the status to the client.
+                    if (transactionType != TransactionType.AUTHORIZE &&
```

Comment:
I am also not clear about whether we should ignore `TransactionType.AUTHORIZE` here. I am tempted to say it would be better to keep the code generic (if indeed it works for all use cases), but i also understand your point about already having some logic for `TransactionType.AUTHORIZE` . I 'd like to have a conversation with @pierre to decide what is the best course of action.